Source/JavaScriptCore/runtime/JSONObject.cpp

```diff
@@ -353,26 +356,27 @@ Stringifier::StringifyResult Stringifier::appendStringifiedValue(StringBuilder&
         return StringifyFailedDueToUndefinedOrSymbolValue;
 
     if (value.isNull()) {
-        builder.append("null");
+        builder.append("null"_s);
         return StringifySucceeded;
     }
 
-    value = unwrapBoxedPrimitive(m_globalObject, value);
-
-    RETURN_IF_EXCEPTION(scope, StringifyFailed);
+    if (value.isObject()) {
+        value = unwrapBoxedPrimitive(m_globalObject, asObject(value));
+        RETURN_IF_EXCEPTION(scope, StringifyFailed);
+    }
```

Comment:
Can't this just be the following, given your `unwrapBoxedPrimitive` definition above? 
```suggestion
    value = unwrapBoxedPrimitive(m_globalObject, value);
    RETURN_IF_EXCEPTION(scope, StringifyFailed);
```

Reply:
That’s an optimization to avoid doing the `RETURN_IF_EXCEPTION` logic when it’s not needed. Not sure how effective it is, but I think it’s a tiny speedup.